tests/HasTagsScopesTest.php

```diff
@@ -115,4 +122,36 @@ public function it_provides_as_scope_to_get_all_models_that_have_all_of_the_give
 
         $this->assertEquals(['model5'], $testModels->pluck('name')->toArray());
     }
+
+    /** @test */
+    public function it_provides_as_scope_to_get_all_models_that_have_all_of_the_given_tags_with_type_and_a_specific_locale()
+    {
+        $testModels = TestModel::withAllTags(['tagG'], 'translatedTag' ,'fr')->get();
+
+        $this->assertEquals(['model7'], $testModels->pluck('name')->toArray());
+    }
+
+    /** @test */
+    public function it_provides_as_scope_to_get_all_models_that_have_any_of_the_given_tags_and_a_specific_locale()
+    {
+        $testModels = TestModel::withAnyTags(['tagG'], 'translatedTag' ,'fr')->get();
+
+        $this->assertEquals(['model7'], $testModels->pluck('name')->toArray());
+    }
+
+    /** @test */
+    public function it_provides_as_scope_to_get_all_models_that_have_any_of_the_given_tags_with_any_type_and_a_specific_locale()
+    {
+        $testModels = TestModel::withAnyTagsOfAnyType(['tagG'], 'fr')->get();
+
+        $this->assertEquals(['model7'], $testModels->pluck('name')->toArray());
+    }
+
+    /** @test */
+    public function it_provides_as_scope_to_get_all_models_that_have_all_of_the_given_tags_with_any_type_and_a_specific_locale()
+    {
+        $testModels = TestModel::withAllTagsOfAnyType(['tagG'], 'fr')->get();
+
+        $this->assertEquals(['model7'], $testModels->pluck('name')->toArray());
+    }
```

Comment:
Also add tests for the case where you try to get a locale, but no locale is used for a tag. What happens than?

Reply:
I added some tests but I'm not sure it doesn't duplicate the existing ones.